### PR TITLE
ARROW-6429: [Integration] Adding patch to fix Spark compilation for Integrition tests

### PR DESCRIPTION
**Pull request by BryanCutler**
Arrow Java Writer now requires an IpcOption for some APIs, this patch fixes the compilation to run Spark Integration tests.

### CODE REVIEW

File: integration/spark/runtest.sh
Comment (BryanCutler) [inline integration/spark/runtest.sh]: @kszucs I thought this would be better in the Dockerfile after checking out spark, but for some reason the patch didn't seem to apply. I was trying this and it seemed to run, but the file wasn't patched
```
COPY integration/spark/spark-arrow-java-IpcOption.patch /spark/
RUN patch -d /spark -p1 -i /tmp/spark-arrow-java-IpcOption.patch
```
Any ideas?

**Comment by BryanCutler** [comment]
cc @kszucs @wesm

File: integration/spark/runtest.sh
Comment (BryanCutler) [reply integration/spark/runtest.sh]: Nevermind, it seems to work now. I'll change it..

**Comment by kszucs** [comment]
@ursabot crossbow submit docker-spark-integration

**Comment by kszucs** [comment]
@BryanCutler seems like the build filed with a timeout `Too long with no output (exceeded 10m0s)`

Any ideas how could we speed up the spark integration test a bit?

**Comment by BryanCutler** [comment]
> Any ideas how could we speed up the spark integration test a bit?

@kszucs , I noticed the Java builds for Arrow and Spark are unusually slow. I'm not sure why, but I'll take a look at the settings. Also, the pyspark tests run twice, once against python and python3.6, which I think are the same in this image. So we can just explicitly test one and that will save a couple minutes I think.

**Comment by BryanCutler** [comment]
Once #5471 is merged, we should be able to get a pass here (as long as no timeout).

File: integration/spark/Dockerfile
Comment (BryanCutler) [inline integration/spark/Dockerfile]: This is the minimum version used by Spark, so setting this here will prevent Spark from downloading it during the build phase

File: integration/spark/runtest.sh
Comment (BryanCutler) [inline integration/spark/runtest.sh]: Spark will look for and test separately against installed python versions, so setting this will make sure to run the tests once on the default python

**Comment by BryanCutler** [comment]
@kszucs , I made a couple small adjustments but I think the reason for the timeout is that Spark can take a long time during assembly, which downloads and assembles all required dependencies. I don't think there is anyway to avoid this since we need to test pyspark, but perhaps there is someway to cache better.

Since this could take a long time and we limit the output during the build to just warnings, we end up hitting the "too long without output" timeout. Is it possible to increase this by setting `no_output_timeout` to something like 30min?

**Comment by kszucs** [comment]
@BryanCutler seems like we can increase that timeout https://support.circleci.com/hc/en-us/articles/360007188574-Build-has-hit-timeout-limit

I'm going to try to push and pull images before and after the build and run to spare the docker-compose build time.

**Comment by BryanCutler** [comment]
Thanks @kszucs , so will you be able to adjust that timeout or can I do it somewhere from this PR? I'm not sure where the config file is..  I'll try to trigger another test since it should pass now if it doesn't timeout.

**Comment by BryanCutler** [comment]
@ursabot crossbow submit docker-spark-integration

**Comment by kszucs** [comment]
@BryanCutler #5485 should speed up the things a bit, and sets no_output_timeout to an hour.

**Comment by BryanCutler** [comment]
@kszucs so should I wait until #5485 is merged to see if we can get a pass here, or go ahead and merge this first?

**Comment by kszucs** [comment]
If you've tried it locally then we can go ahead and merge this.

**Review by kszucs — APPROVED** [review]
The patch LGTM.

**Comment by BryanCutler** [comment]
merged to master, thanks for reviewing @kszucs !

**Pull request closed without merging**
